contracts/MarketContract.sol

```diff
@@ -157,10 +157,12 @@ contract MarketContract is Creatable {
         OrderLib.Order memory order = address(this).createOrder(orderAddresses, unsignedOrderValues, orderQty);
         require(MKT_TOKEN.isUserEnabledForContract(this, order.maker));
 
+        // do not allow trade without a taker
+        require(order.taker != 0);
```

Comment:
order.taker can be a null address, If I place an order, and would like anyone to be able to fill it then I would leave the address null, so this require statement doesn't work here.